share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,854 @@
+```
+def aveI = \x\y (x+y)/2
+```
+
+* Function aveR(averageOfReal) takes 2 arguments of type double and divides the result by 2.
+```
+def aveR = \x\y (x +. y) /. 2.0
+```
+
+* Function aveM(Modulus) adds 2 values and provides the remainder.
+```
+def aveM = \x\y (x+y) % 2
+```
+
+### Wake Built-In Functions
+Few of the commonly used built-in functions in this tutorial is given below
+#### Operation on String
+* `explode` breaks a string into a list of string character.
+* `strlen` returns the length of the given string.
+* `cat` concatenates the elements in list of string to a string. 
+
+* strlen returns the length of the given string.
+```
+wake -x 'strlen "abcde"'
+```
+
+#### Operation on List
+* `take` takes first `n` elements in the list.
+* `drop` drops first `n` elements in the list.
+* `reverse` reverses the elements in the list.
+* `head` returns the first element in the list.
+* `tail` returns the list by removing the `head`.
+* `prepend` places the given data at the begining(head) of the given list.
+* `append` appends the given data at the end of the given list.
+* `++` concatenates any number of list into a single list.
+* `len` returns the length of the given list.
+* `flatten` converts the list of list into a list.
+* `mapFlat` applies a function to each element in the list and builds a new list from the resulting elements.
+* `filter`  applies a function and builds a new list from it.
+* `zip` takes 2 lists and converts them to list of Pairs.
+
+* Function duplicate concantenates or copies a given string twice.
+```
+def duplicate s  ="{s}{s}"
+```
+#### Miscellaneous Built-In Functions 
+* integerToUnicode is a `wake built-in function` which takes the ASCII input and provides its equivalent character.
+```
+wake -x 'integerToUnicode 99'
+```
+
+* unicodeToInteger is a `wake built-in function` which gives the ASCII value of the given character.
+```
+wake -x 'unicodeToInteger "a"'
+```
+
+### List
+A List is a sequence of items of same type. Empty list returns `Nil`. `head` returns the first element in the list. `tail` returns the rest of the list without the `head`.
+
+* Example creates a list of given elements.
+```
+def listcreate xx = xx,xx,xx,Nil
+```
+ 
+* Example shows how to prepend the list of integers with the given value.
+```
+def d = 1,2,3,Nil
+def listprepend data1 = prepend data1 d
+```
+
+* `Note: Parenthesis in wake is used to enforce a specific order of evaluation(precedence of operators) in expressions.`
+* concantenates 2 lists using ++ operator
+```
+wake -x '(1,2,3,Nil) ++ (4,5,6,Nil)'
+```
+### Option
+`Option` is a data type in wake. It will either return `Some x` which is a value `x` or `None` which means nothing.
+
```

Comment:
how do you create a Some x or a None? I couldn't figure out how to *create* an option vs matching against one. It's not:

`Some 3`.

Is it `3.Some`?

Reply:
It is `Some `...